dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/controller/DPPPTController.java

```diff
@@ -124,6 +125,44 @@ public DPPPTController(DPPPTDataService dataService, EtagGeneratorInterface etag
 		return ResponseEntity.ok().build();
 	}
 
+	@CrossOrigin(origins = { "https://editor.swagger.io" })
+	@PostMapping(value = "/exposedlist")
+	public @ResponseBody ResponseEntity<String> addExposee(@Valid @RequestBody ExposeeRequestList exposeeRequests,
+			@RequestHeader(value = "User-Agent", required = true) String userAgent,
+			@AuthenticationPrincipal Object principal) throws InvalidDateException {
+		long now = System.currentTimeMillis();
+		if (!this.validateRequest.isValid(principal)) {
+			return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+		}
+		List<Exposee> exposees = new ArrayList<>();
+		for(var exposedKey : exposeeRequests.getExposedKeys()) {
+			if (!isValidBase64(exposedKey.getKey())) {
+				return new ResponseEntity<>("No valid base64 key", HttpStatus.BAD_REQUEST);
+			}
+
+			Exposee exposee = new Exposee();
+			exposee.setKey(exposedKey.getKey());
+			long keyDate = this.validateRequest.getKeyDate(principal, exposedKey);
+
+			exposee.setKeyDate(keyDate);
+			if(!this.validateRequest.isFakeRequest(principal, exposeeRequests)) {
+				
+				exposees.add(exposee);
+			} 
+		}
+		dataService.upsertExposees(exposees, appSource);
```

Comment:
Fake requests trigger an SQL batch update with an empty list here.  Would it make sense to invert the for loop and the if statement? or skipping the SQL query when the list is empty?

Reply:
yes you're right of course :D Forgot to take it out of the loop. Now it should only upsert, if it is not a fake request